databricks/koalas/indexes.py

```diff
@@ -679,13 +679,12 @@ def to_series(self, name: Union[str, Tuple[str, ...]] = None) -> Series:
         scol = self.spark.column
         if name is not None:
             scol = scol.alias(name_like_string(name))
+        elif len(kdf._internal.index_map) == 1:
+            name = self.name
         column_labels = (
             [(SPARK_DEFAULT_SERIES_NAME,)]
             if len(kdf._internal.index_map) > 1
-            else [
-                (SPARK_DEFAULT_SERIES_NAME,) if name is None else name
-                for name in kdf._internal.index_names
-            ]
+            else [(SPARK_DEFAULT_SERIES_NAME,) if name is None else (name,)]
```

Comment:
None of us is very fluent in mypy and type hinting, @itholic. It might be false-positive.

```
databricks/koalas/indexes.py: note: In member "to_series" of class "Index":
databricks/koalas/indexes.py:687: error: List item 0 has incompatible type "Tuple[Union[str, Tuple[str, ...]]]"; expected "Tuple[str]"
```

Reply:
Yeah, we are always doing trial and error. 😄 

How about:

```diff
             name = self.name
         column_labels = (
             [(SPARK_DEFAULT_SERIES_NAME,)]
-            if len(kdf._internal.index_map) > 1
-            else [(SPARK_DEFAULT_SERIES_NAME,) if name is None else (name,)]
-        )
+            if len(kdf._internal.index_map) > 1 or name is None
+            else [name if isinstance(name, tuple) else (name,)]
+        )  # type: List[Tuple[str, ...]]
         internal = kdf._internal.copy(
             column_labels=column_labels, data_spark_columns=[scol], column_label_names=None
         )
```

Reply:
Thanks @HyukjinKwon , @ueshin .

I'll have to try to get familiar with type checking with `mypy`.

Sincerely thanks for your time for addressing this !!